File: c/4-vectors/position-largest-smallest/main.c
// Write a program to insert values ​​into a vector. Print the largest, smallest and their respective positions.

#include <stdio.h>
#include <stdlib.h>

int main()
{

    int i, n, number, largest, posLargest, smallest, posSmallest;
    printf("Enter the number of elements to insert: ");
    scanf("%d", &n);

    int vec[n];
    for (i = 0; i < n; i++) {
        printf("\nEnter a number: ");
        scanf("%d", &number);
        vec[i] = number;
    }

    largest = vec[0];
    posLargest = 0;
    for (i = 1; i < n; i++) {
        if (vec[i] > largest) {
            largest = vec[i];
            posLargest = i;
        }
    }

    smallest = vec[0];
    posSmallest = 0;
    for (i = 1; i < n; i++) {
        if (vec[i] < smallest) {
            smallest = vec[i];
            posSmallest = i;
        }
    }

    printf("\nThe largest number inserted in the vector is %d and is in position %d.\n"
    "The smallest number inserted in the vector is %d and is in position %d.\n", largest, posLargest, smallest, posSmallest);

    return 0;
}

File: python/4-vectors/position-largest-smallest/main.py
# Write a program to insert values into a vector.
# Print the largest, smallest and their respective positions.

n = int(input("Enter the number of elements to insert: "))

vec = []
for i in range(n):
    number = int(input("\nEnter a number: "))
    vec.append(number)

largest = vec[0]
pos_largest = 0
for i in range(1, n):
    if vec[i] > largest:
        largest = vec[i]
        pos_largest = i

smallest = vec[0]
pos_smallest = 0
for i in range(1, n):
    if vec[i] < smallest:
        smallest = vec[i]
        pos_smallest = i

print(f"\nThe largest number inserted in the vector is {largest} and is in position {pos_largest}.\n\
The smallest number inserted in the vector is {smallest} and is in position {pos_smallest}.")